packages/cozy-doctypes/src/File.js

```diff
@@ -23,6 +23,52 @@ class CozyFile extends Document {
     const parentDirectoryPath = trimEnd(parentDir.path, '/')
     return `${parentDirectoryPath}/${name}`
   }
+
+  /**
+   * Move file to destination.
+   *
+   * @param   {string} fileId               - The file's id (required)
+   * @param   {object} destination
+   * @param   {string} destination.folderId - The destination folder's id (required)
+   * @param   {string} destination.path     - The file's path after the move (optional)
+   * @param   {string} force                - Whether we should overwrite the destination in case of conflict (defaults to false)
+   * @returns {Promise}                     - A promise that returns the move action response and the deleted file id (if any) if resolved or an Error if rejected
+   *
+   */
+  static async move(fileId, destination, force = false) {
+    const filesCollection = this.cozyClient.collection('io.cozy.files')
+    try {
+      const resp = await filesCollection.updateFileMetadata(fileId, {
+        dir_id: destination.folderId
+      })
+
+      return {
+        moved: resp.data,
+        deleted: null
+      }
+    } catch (e) {
+      if (e.status === 409 && force) {
+        if (!destination.path || typeof destination.path !== 'string') {
+          throw new Error(
+            'You must provide a destination path if you use force option.'
```

Comment:
Follow-up from https://github.com/cozy/cozy-libs/pull/460?email_source=notifications&email_token=AAADXJ244USSR2HQ663P5IDPXP5AHA5CNFSM4HP43U22YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZYZZWY#discussion_r287829820


I still don't understand why we require a path to force the overwrite. If the fileId has already has a name and the folderId already has a path, the destination path could be built from the two. Am I wrong?

Reply:
Si sur un `get(fileId)` on est capable de récupérer le path complet du file alors oui on doit pouvoir s'en passer. A vérifier

Reply:
I can get the folder's path from the `folderId` and get the name from the `fileId` (because in case of conflict, we know the file name is the same) and so build the destination path but I wanted to avoid unnecessary queries since I have to calculate the path in drive before calling `move`. It's probably not a good reason to complexify this method's signature though.

Reply:
If so, you may leave the optional parameter as a way to enhance performance but not require it

Reply:
Good idea, it's done